File: ViewModels/CellViewModel.cs
using System.Windows.Media;
using Crossword.Infrastructure;

namespace Crossword.ViewModels;

public class CellViewModel : ViewModelBase
{
    private Brush _background = Brushes.Black;
    private string? _content;
    private int _x;
    private int _y;
    public bool IsPreview { get; set; }

    public int X
    {
        get => _x;
        set => SetProperty(ref _x, value);
    }

    public int Y
    {
        get => _y;
        set => SetProperty(ref _y, value);
    }

    public int DisplayX => IsPreview ? (X - 1) * PreviewCellSize : X * CellSize;
    public int DisplayY => IsPreview ? (Y - 1) * PreviewCellSize : Y * CellSize;

    public Brush Background
    {
        get => _background;
        set => SetProperty(ref _background, value);
    }

    public string? Content
    {
        get => _content;
        set => SetProperty(ref _content, value);
    }

    public static int CellSize => 30;
    public static int PreviewCellSize => 5;
    public int Width => IsPreview ? PreviewCellSize : CellSize;
    public int Height => IsPreview ? PreviewCellSize : CellSize;
}

File: crossword_python/viewmodels/CellViewModel.py
from crossword_python.core.viewmodel_base import ViewModelBase

class CellViewModel(ViewModelBase):
    CELL_SIZE = 30
    PREVIEW_CELL_SIZE = 5

    def __init__(self):
        super().__init__()
        self._background = "black"
        self._content = None
        self._x = 0
        self._y = 0
        self.is_preview = False

    @property
    def x(self) -> int:
        return self._x

    @x.setter
    def x(self, value: int):
        self.set_property('_x', value)

    @property
    def y(self) -> int:
        return self._y

    @y.setter
    def y(self, value: int):
        self.set_property('_y', value)

    @property
    def display_x(self) -> int:
        return (self.x - 1) * self.PREVIEW_CELL_SIZE if self.is_preview else self.x * self.CELL_SIZE

    @property
    def display_y(self) -> int:
        return (self.y - 1) * self.PREVIEW_CELL_SIZE if self.is_preview else self.y * self.CELL_SIZE

    @property
    def background(self) -> str:
        return self._background

    @background.setter
    def background(self, value: str):
        if value == "black":
            self.content = None
        self.set_property('_background', value)

    @property
    def content(self) -> str | None:
        return self._content

    @content.setter
    def content(self, value: str | None):
        self.set_property('_content', value)

    @property
    def width(self) -> int:
        return self.PREVIEW_CELL_SIZE if self.is_preview else self.CELL_SIZE

    @property
    def height(self) -> int:
        return self.PREVIEW_CELL_SIZE if self.is_preview else self.CELL_SIZE